cmake/multi_image.cmake

```
@@ -0,0 +1,86 @@
+#
+# Copyright (c) 2019 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+#
+
+function(zephyr_add_external_image name sourcedir)
+  string(TOUPPER ${name} NAME)
+  set(${NAME}_CMAKE_BINARY_DIR ${CMAKE_BINARY_DIR}/${name})
+  set(${NAME}_PROJECT_BINARY_DIR ${CMAKE_BINARY_DIR}/${name}/zephyr)
+
+  file(MAKE_DIRECTORY ${${NAME}_CMAKE_BINARY_DIR})
+  message("=== child image ${name} begin ===")
```

If changing this from `subimage` anyway then I would actually prefer to use the CMake naming style and call this
```suggestion
  message("=== external project ${name} begin ===")
```
as this will provide more direct information to users familiar with CMake.

---

Child image is established terminology and is more precise than external project.

https://developer.nordicsemi.com/nRF_Connect_SDK/doc/1.0.0/nrf/ug_multi_image.html#multi-image-builds

---

yep, I noticed AFTER i found the user guide.
Although I still dislike the term in CMake context, (as seen from inside CMake it is an external project), then because it is already used in the user guide, then let's keep it.